framework/doc/content/source/controls/WebServerControl.md

```diff
@@ -0,0 +1,116 @@
+# WebServerControl
+
+The `WebServerControl` object is designed to allow an external process to control a MOOSE solve. It works by starting up a webserver on the chosen [!param](/Controls/WebServerControl/port) and exposing a REST API. As with all REST APIs the input and output are both JSON.
+
+It can then be managed via the [MooseControl/index.md] python utility.
+
+## API
+
+The `WebServerControl` presents REST endpoints to help control the solve.
+
+For the endpoints that follow, if the HTTP status code received is a non-successful code, there may be a response of type `application/json` of the form:
+
+```language=json
+{
+  "error": "<ERROR>"
+}
+```
+
+where `<ERROR>` is a useful error message pertaining to the error that was encountered. The response is not guaranteed to contain `application/json` data, but it will when more context is available.
+
+The REST endpoints are as follows:
+
+### `waiting`
+
+Used to determine of MOOSE is currently waiting within the `WebServerControl`, i.e., that `Control` is waiting for external input. The rest of the endpoints that follow require that the control is currently waiting for input.
+
+Interact with this endpoint by a `GET` request to `/waiting` without any data. If the control is currently waiting, the response will be of type `application/json` of the form:
+
+```language=json
+{
+  "waiting": True
+  "execute_on_flag": "<EXEC_ON_FLAG>"
+}
+```
+
+where `<EXEC_ON_FLAG>` is the current execution flag. If the control is not currently waiting, the response will be a response of type `application/json` of the form:
+
+```language=json
+{
+  "waiting": False
+}
+```
+
+This endpoint can be accessed via the [MooseControl/index.md] python utility via the following methods:
+
+- `wait()`: Waits for the control to be waiting
+- `getWaitingFlag()`: Gets the current flag that the control is waiting on, if any
+- `isWaiting()`: Whether or not the control is currently waiting
+
+### `get/postprocessor`
+
+Used to obtain the value of a postprocessor. The control must be waiting in order to access this endpoint.
+
+Interact with this endpoint by a `POST` request to `/get/postprocessor` with the following `application/json` data:
+
+```language=json
+{
+  "name": "<NAME>"
+}
+```
+
+where `<NAME>` is the name of the postprocessor whose value you wish to receive. The response will be of the form:
+
+```language=json
+{
+  "value": <VALUE>
+}
+```
+
+### `set/controllable`
```

Comment:
Would be neat to have an endpoint that gets a list of all controllable parameters and their types as well as all available postprocessors (to enable writing a generic control interface)

Reply:
I agree. I thought about this, but currently the controllable registry is in awful shape... and I don't have the scope/time for it at the moment for that to work properly